SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -330,29 +333,37 @@ void ComponentFactory::AddProperty(Schema_Object* Object, Schema_FieldId FieldId
 	}
 }
 
-TArray<FWorkerComponentData> ComponentFactory::CreateComponentDatas(UObject* Object, const FClassInfo& Info, const FRepChangeState& RepChangeState, const FHandoverChangeState& HandoverChangeState)
+TArray<FWorkerComponentData> ComponentFactory::CreateComponentDatas(UObject* Object, const FClassInfo& Info, const FRepChangeState& RepChangeState, const FHandoverChangeState& HandoverChangeState, uint32& OutBytesWritten)
 {
 	TArray<FWorkerComponentData> ComponentDatas;
 
+	OutBytesWritten = 0;
```

Comment:
Not sure we should do this

Reply:
Why not? All functions inside ComponentFactory set OutBytesWritten instead of adding to it.

Reply:
I think it's fine. I would expect that I get a set result rather than an additive result

Reply:
Oh boy... you won't like my next commit.